### PR TITLE
Tao support macos

### LINKED ISSUE #491: Fix build procedure on README.md
The description of the repos and build process needs update after the great merge of Feb 17.

### PR DESCRIPTION
**Pull request by ptpt**
This PR adds initial support for macOS (only for development, no production).

Still 3 tests failed. `test/json` failed because at line https://github.com/valhalla/valhalla/blob/master/test/json.cc#L79-L80 `"\"status\": \"0\"," != "\"status\": \"0\""`. Can anyone help?

```
======================================
   valhalla 2.0.7: ./test-suite.log
======================================

# TOTAL: 77
# PASS:  74
# SKIP:  0
# XFAIL: 0
# FAIL:  3
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: test/json
===============

=== Testing json ===
               TestJsonSerialize  [FAIL: Wrong json!]
=== Failed 1 tests ===
FAIL test/json (exit status: 1)

FAIL: test/narrativebuilder
===========================

=== Testing narrativebuilder ===
 TestFormRampStraightInstruction  [PASS]
    TestFormRampRightInstruction  [PASS]
     TestFormRampLeftInstruction  [PASS]
    TestFormExitRightInstruction  [PASS]
     TestFormExitLeftInstruction  [PASS]
TestFormVerbalPostTransitionInstruction  [PASS]
TestBuildStartInstructions_0_miles_en_US  [PASS]
TestBuildStartInstructions_1_miles_en_US  [PASS]
TestBuildStartInstructions_2_miles_en_US  [PASS]
TestBuildStartInstructions_4_miles_en_US  [PASS]
TestBuildStartInstructions_5_miles_en_US  [PASS]
TestBuildStartInstructions_6_miles_en_US  [PASS]
TestBuildStartInstructions_8_miles_en_US  [PASS]
TestBuildStartInstructions_9_miles_en_US  [PASS]
TestBuildStartInstructions_9_unnamed_walkway_miles_en_US  [PASS]
TestBuildStartInstructions_10_miles_en_US  [PASS]
TestBuildStartInstructions_16_miles_en_US  [PASS]
TestBuildStartInstructions_17_miles_en_US  [PASS]
TestBuildStartInstructions_17_unnamed_cycleway_miles_en_US  [PASS]
TestBuildStartInstructions_17_unnamed_mountain_bike_trail_miles_en_US  [PASS]
TestBuildStartInstructions_18_miles_en_US  [PASS]
TestBuildStartInstructions_0_kilometers_en_US  [PASS]
TestBuildStartInstructions_1_kilometers_en_US  [PASS]
TestBuildStartInstructions_2_kilometers_en_US  [PASS]
TestBuildDestinationInstructions_0_miles_en_US  [PASS]
TestBuildDestinationInstructions_1_miles_en_US  [PASS]
TestBuildDestinationInstructions_2_miles_en_US  [PASS]
TestBuildDestinationInstructions_3_miles_en_US  [PASS]
TestBuildBecomesInstructions_0_miles_en_US  [PASS]
TestBuildContinueInstructions_0_miles_en_US  [PASS]
TestBuildContinueInstructions_1_miles_en_US  [PASS]
TestBuildTurnInstructions_0_miles_en_US  [PASS]
TestBuildTurnInstructions_1_miles_en_US  [PASS]
TestBuildTurnInstructions_1_miles_cs_CZ  [PASS]
TestBuildTurnInstructions_1_miles_de_DE  [PASS]
TestBuildTurnInstructions_1_miles_it_IT  [PASS]
TestBuildTurnInstructions_2_miles_en_US  [PASS]
TestBuildTurnInstructions_3_miles_en_US  [PASS]
TestBuildSharpInstructions_0_miles_en_US  [PASS]
TestBuildSharpInstructions_1_miles_en_US  [PASS]
TestBuildSharpInstructions_2_miles_en_US  [PASS]
TestBuildSharpInstructions_0_miles_en_US  [PASS]
TestBuildBearInstructions_0_miles_en_US  [PASS]
TestBuildBearInstructions_1_miles_en_US  [PASS]
TestBuildBearInstructions_2_miles_en_US  [PASS]
TestBuildBearInstructions_3_miles_en_US  [PASS]
TestBuildUturnInstructions_0_miles_en_US  [PASS]
TestBuildUturnInstructions_1_miles_en_US  [PASS]
TestBuildUturnInstructions_2_miles_en_US  [PASS]
TestBuildUturnInstructions_3_miles_en_US  [PASS]
TestBuildUturnInstructions_4_miles_en_US  [PASS]
TestBuildUturnInstructions_5_miles_en_US  [PASS]
TestBuildRampStraight_0_miles_en_US  [PASS]
TestBuildRampStraight_1_miles_en_US  [PASS]
TestBuildRampStraight_2_miles_en_US  [PASS]
TestBuildRampStraight_3_miles_en_US  [PASS]
TestBuildRampStraight_4_miles_en_US  [PASS]
     TestBuildRamp_0_miles_en_US  [PASS]
     TestBuildRamp_1_miles_en_US  [PASS]
     TestBuildRamp_2_miles_en_US  [PASS]
     TestBuildRamp_3_miles_en_US  [PASS]
     TestBuildRamp_4_miles_en_US  [PASS]
     TestBuildRamp_5_miles_en_US  [PASS]
     TestBuildRamp_6_miles_en_US  [PASS]
     TestBuildRamp_7_miles_en_US  [PASS]
     TestBuildRamp_8_miles_en_US  [PASS]
     TestBuildRamp_9_miles_en_US  [PASS]
     TestBuildExit_0_miles_en_US  [PASS]
     TestBuildExit_1_miles_en_US  [PASS]
     TestBuildExit_2_miles_en_US  [PASS]
     TestBuildExit_3_miles_en_US  [PASS]
     TestBuildExit_4_miles_en_US  [PASS]
     TestBuildExit_5_miles_en_US  [PASS]
     TestBuildExit_6_miles_en_US  [PASS]
     TestBuildExit_7_miles_en_US  [PASS]
     TestBuildExit_8_miles_en_US  [PASS]
    TestBuildExit_10_miles_en_US  [PASS]
    TestBuildExit_12_miles_en_US  [PASS]
    TestBuildExit_14_miles_en_US  [PASS]
     TestBuildKeep_0_miles_en_US  [PASS]
     TestBuildKeep_1_miles_en_US  [PASS]
     TestBuildKeep_2_miles_en_US  [PASS]
     TestBuildKeep_3_miles_en_US  [PASS]
     TestBuildKeep_4_miles_en_US  [PASS]
     TestBuildKeep_5_miles_en_US  [PASS]
     TestBuildKeep_6_miles_en_US  [PASS]
     TestBuildKeep_7_miles_en_US  [PASS]
TestBuildKeepToStayOn_0_miles_en_US  [PASS]
TestBuildKeepToStayOn_1_miles_en_US  [PASS]
TestBuildKeepToStayOn_2_miles_en_US  [PASS]
TestBuildKeepToStayOn_3_miles_en_US  [PASS]
    TestBuildMerge_0_miles_en_US  [PASS]
  TestBuildMerge_1_1_miles_en_US  [PASS]
  TestBuildMerge_1_2_miles_en_US  [PASS]
TestBuildEnterRoundabout_0_miles_en_US  [PASS]
TestBuildEnterRoundabout_1_miles_en_US  [PASS]
TestBuildExitRoundabout_0_miles_en_US  [PASS]
TestBuildExitRoundabout_1_miles_en_US  [PASS]
TestBuildExitRoundabout_2_miles_en_US  [PASS]
TestBuildEnterFerry_0_miles_en_US  [PASS]
TestBuildEnterFerry_1_miles_en_US  [PASS]
TestBuildEnterFerry_2_miles_en_US  [PASS]
TestBuildExitFerry_0_miles_en_US  [PASS]
TestBuildExitFerry_1_miles_en_US  [PASS]
TestBuildExitFerry_2_miles_en_US  [PASS]
TestBuildExitFerry_4_miles_en_US  [PASS]
TestBuildExitFerry_5_miles_en_US  [PASS]
TestBuildExitFerry_6_miles_en_US  [PASS]
TestBuildExitFerry_8_miles_en_US  [PASS]
TestBuildExitFerry_9_miles_en_US  [PASS]
TestBuildExitFerry_10_miles_en_US  [PASS]
TestBuildExitFerry_16_miles_en_US  [PASS]
TestBuildExitFerry_17_miles_en_US  [PASS]
TestBuildExitFerry_18_miles_en_US  [PASS]
TestBuildTransitConnectionStart_0_miles_en_US  [PASS]
TestBuildTransitConnectionStart_1_miles_en_US  [PASS]
TestBuildTransitConnectionStart_2_miles_en_US  [PASS]
TestBuildTransitConnectionTransfer_0_miles_en_US  [PASS]
TestBuildTransitConnectionTransfer_1_miles_en_US  [PASS]
TestBuildTransitConnectionTransfer_2_miles_en_US  [PASS]
TestBuildTransitConnectionDestination_0_miles_en_US  [PASS]
TestBuildTransitConnectionDestination_1_miles_en_US  [PASS]
TestBuildTransitConnectionDestination_2_miles_en_US  [PASS]
TestBuildTransit_0_train_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:02 AM from 8 St - NYU.  |  produced: Depart: 08:02 from 8 St - NYU.]
  TestBuildTransit_0_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:02 AM from 8 St - NYU.  |  produced: Depart: 08:02 from 8 St - NYU.]
TestBuildTransit_1_cable_car_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:03 AM from Hyde St & Bay St.  |  produced: Depart: 08:03 from Hyde St & Bay St.]
TestBuildTransit_1_stop_count_1_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:06 AM from Union St.  |  produced: Depart: 08:06 from Union St.]
TestBuildTransit_1_stop_count_2_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:05 AM from 28 St.  |  produced: Depart: 08:05 from 28 St.]
TestBuildTransit_1_stop_count_4_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:02 AM from 8 St - NYU.  |  produced: Depart: 08:02 from 8 St - NYU.]
TestBuildTransit_1_stop_count_8_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:11 AM from Flushing Av.  |  produced: Depart: 08:11 from Flushing Av.]
TestBuildTransit_1_stop_count_1_miles_cs_CZ  [PASS]
TestBuildTransit_1_stop_count_2_miles_cs_CZ  [PASS]
TestBuildTransit_1_stop_count_4_miles_cs_CZ  [PASS]
TestBuildTransit_1_stop_count_8_miles_cs_CZ  [PASS]
TestBuildTransitTransfer_0_no_name_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:02 AM from 8 St - NYU.  |  produced: Depart: 08:02 from 8 St - NYU.]
TestBuildTransitTransfer_0_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:02 AM from 8 St - NYU.  |  produced: Depart: 08:02 from 8 St - NYU.]
TestBuildTransitTransfer_1_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:02 AM from 8 St - NYU.  |  produced: Depart: 08:02 from 8 St - NYU.]
TestBuildTransitRemainOn_0_no_name_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:02 AM from 8 St - NYU.  |  produced: Depart: 08:02 from 8 St - NYU.]
TestBuildTransitRemainOn_0_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:02 AM from 8 St - NYU.  |  produced: Depart: 08:02 from 8 St - NYU.]
TestBuildTransitRemainOn_1_miles_en_US  [FAIL: Incorrect maneuver depart_instruction - expected: Depart: 8:02 AM from 8 St - NYU.  |  produced: Depart: 08:02 from 8 St - NYU.]
TestBuildPostTransitConnectionDestination_0_miles_en_US  [PASS]
TestBuildPostTransitConnectionDestination_1_miles_en_US  [PASS]
TestBuildPostTransitConnectionDestination_2_miles_en_US  [PASS]
TestBuildPostTransitConnectionDestination_4_miles_en_US  [PASS]
TestBuildPostTransitConnectionDestination_5_miles_en_US  [PASS]
TestBuildPostTransitConnectionDestination_6_miles_en_US  [PASS]
TestBuildPostTransitConnectionDestination_8_miles_en_US  [PASS]
TestBuildPostTransitConnectionDestination_9_miles_en_US  [PASS]
TestBuildPostTransitConnectionDestination_10_miles_en_US  [PASS]
TestBuildPostTransitConnectionDestination_16_miles_en_US  [PASS]
TestBuildPostTransitConnectionDestination_17_miles_en_US  [PASS]
TestBuildPostTransitConnectionDestination_18_miles_en_US  [PASS]
TestBuildVerbalMultiCue_0_miles_en_US  [PASS]
=== Failed 13 tests ===
FAIL test/narrativebuilder (exit status: 1)

FAIL: test/util_odin
====================

=== Testing util ===
          test_supported_locales  [FAIL: Locale not found for: hi_IN.UTF-8]
                test_get_locales  [PASS]
                       test_time  [FAIL: Incorrect Time: 23:59 ---> 11:59 PM for locale: en_US.UTF-8]
                       test_date  [FAIL: Incorrect Date: 2014/01/01 ---> 1.1.2014 for locale: cs_CZ.UTF-8]
=== Failed 3 tests ===
FAIL test/util_odin (exit status: 1)
```

fixes #491

### CODE REVIEW

**Comment by ptpt** [comment]
I think it's ready to merge if the failing tests don't affect development that much.

File: README.md [hunk 1]
```diff
@@ -74,11 +74,18 @@ Valhalla uses the [GNU Build System](http://www.gnu.org/software/automake/manual
...
+
 And then run to install it:
 
+    git submodule update --init --recursive
     ./autogen.sh
+    # on macOS you need to tell linkers how to reach home-brewed sqlite3
+    # export LDFLAGS="-L/usr/local/opt/sqlite/lib/ -lsqlite3"
     ./configure
-    make test -j$(nproc)
+    make test -j8
```
Comment (kevinkreiser) [inline README.md]: so yeah mac hardware is mostly the same but you cant really do this on any random pc. ones with single or only dual core will basically be fork bombed not to mention the ram requirement is significantly higher when you do this. if mac doesnt have `nproc` hardcoding isnt an option unless we make it `-j1` which i think we all agree is a bad option. maybe we can just do: 

`grep -c processor /proc/cpuinfo`

File: src/valhalla_associate_segments.cc [hunk 2]
```diff
@@ -844,7 +844,7 @@ void edge_association::add_tile(const std::string &file_name) {
     if (!entry.has_marker()) {
       assert(entry.has_segment());
       auto &segment = entry.segment();
-      match_segment(base_id + entry_id, segment);
+      match_segment(base_id + static_cast<uint64_t>(entry_id), segment);
```
Comment (kevinkreiser) [inline src/valhalla_associate_segments.cc]: why not just make `uint64_t` above when declared?

File: README.md [hunk 3]
```diff
@@ -74,11 +74,18 @@ Valhalla uses the [GNU Build System](http://www.gnu.org/software/automake/manual
     #if you plan to compile with python bindings, see below for more info
     sudo apt-get install -y python-all-dev
 
+To install on macOS, you need to install its dependencies with [Homebrew](http://brew.sh):
+
+    brew install autoconf automake libtool protobuf-c boost-python libspatialite pkg-config lua czmq
```
Comment (kevinkreiser) [inline README.md]: this seems to be missing stuff for the prime_server bits specifically. im sure you can just get and compile them but maybe its worth it at this point to get prime_server into brew or valhalla even? a quick google seemed to make it look pretty easy actually (docker-esque)

**Comment by kevinkreiser** [comment]
to fix the json test after we do `read_json` on both of them into `ptree`s we can then just loop over the ptrees and make sure the children are identical, recursively? or we can just ignore this failure because we may as well just move everything to rapidjson in the end and remove ptree and this c++11 style serializer completely.

**Comment by kevinkreiser** [comment]
other than the 2 suggested changes above feel free to merge this!

**Comment by kevinkreiser** [comment]
:ship:

**Pull request merged**